File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let seq= new Set(nums)
        let max=0;
        let count=0
        for(let i of nums){
            if(!seq.has(i-1)){
                let j=i
                 count=1
                while(seq.has(j+1)){
                   
                    j++
                     count++
                }
               
            }
            max=Math.max(count,max)
        }
        return max;
}
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        seq={}
        r,l,maxlen=0,0,0
        while r<len(s):
            if s[r] in seq and l<=seq[s[r]]:
                l=seq[s[r]]+1
        
            seq[s[r]]=r
            maxlen=max(maxlen,r-l+1)
            r+=1
        return maxlen
        